docs/further_useful_features.rst

```diff
@@ -0,0 +1,438 @@
+    )
+    dm.partitions
+
+
+Note that, since 2 dataframes have been provided as input to the function, there are
+4 different files created, even though only 2 different combinations of values of E and
+F are found, ``E=test/F=foo`` and ``E=train/F=foo`` .
+(However, these 4 physical partitions can be read as just the 2 logical partitions by
+using the argument ``concat_partitions_on_primary_index=True`` at reading time).
+
+For datasets consisting of multiple tables, partitioning on
+columns only works if the column exists in both tables and is of the same data type.
+
+For example:
+
+.. ipython:: python
+
+    df.dtypes
+    different_df = pd.DataFrame(
+        {
+            "B": pd.to_datetime(["20130102","20190101"]),
+            "L": [1, 4],
+            "Q": [True, False],
+        }
+    )
+    different_df.dtypes
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "multiple_partitioned_tables",
+        [{"data": {"table1": df, "table2": different_df}}],
+        partition_on='B',
+    )
+
+    dm.partitions
+
+
+Because partitions are physical in nature, it is not possible to modify the
+partitioning scheme of an existing dataset via an update, instead, the dataset
+would have to be re-created.
+
+.. note:: Under the hood, partitions are structurally identical to each other and each partition
+    is made up of a collection of files containing the subset of data of each table
+    belonging to that partition.
+
+Secondary Indices
+-----------------
+
+The ability to build and maintain secondary indices are an additional ability
+provided by ``kartothek``. Secondary indices are `similar` to partitions in the
+sense that they allow faster access to subsets of data. The main difference
+between them is that while partitioning actually creates separate partitions based
+on column values, secondary indices are simply python dictionaries mapping column
+values and the partitions that rows with them can be found in.
+
+.. note::
+
+    The examples we've looked at so far have mostly used functions from the ``eager``
+    backend. As noted earlier, the ``iter`` backend executes operations on the dataset
+    on a per-partition basis and accordingly data inputs are expected to be generators.
+    Although using other iterables such as lists also works, doing so is counter
+    to the intent of the ``iter`` backend (lists would be appropriate in ``eager``).
+
+Writing a dataset with a secondary index:
+
+.. ipython:: python
+
+    from kartothek.io.iter import store_dataframes_as_dataset__iter
+
+    # "Generate" 5 dataframes
+    df_gen = (
+        pd.DataFrame(
+            {
+                "date": pd.Timestamp(f"2020-01-0{i}"),
+                "X": np.random.choice(10, 10),
+            }
+        )
+        for i in range(1, 6)
+    )
+
+    dm = store_dataframes_as_dataset__iter(
+        df_gen,
+        store_factory,
+        "secondarily_indexed",
+        partition_on = "date",
```

Comment:
whitespaces between operator. Should be `partition_on="date"`

Reply:
It may be useful to add a pre-commit hook to run black on docs using https://github.com/asottile/blacken-docs in the future